synapse/handlers/e2e_room_keys.py

```diff
@@ -58,7 +61,9 @@ async def get_room_keys(
         version: str,
         room_id: Optional[str] = None,
         session_id: Optional[str] = None,
-    ) -> List[JsonDict]:
+    ) -> Dict[
+        Literal["rooms"], Dict[str, Dict[Literal["sessions"], Dict[str, RoomKey]]]
+    ]:
```

Comment:
The previous annotation was incorrect. The caller and tests expect a JsonDict to be returned:

https://github.com/matrix-org/synapse/blob/0cc3bf97b4399234cf20f52ae4c09d03661225ff/synapse/rest/client/room_keys.py#L199-L223
https://github.com/matrix-org/synapse/blob/0cc3bf97b4399234cf20f52ae4c09d03661225ff/tests/handlers/test_e2e_room_keys.py#L404-L409

Reply:
Any idea why mypy didn't spot this? I guess the tests aren't checked, but I'd hope room_keys was

Reply:
It looks like `e2e_room_keys_handler.get_room_keys` has a `@trace` decorator, which isn't annotated

Reply:
Interesting! I think @H-Shay is taking a look at annotating that file, maybe this will show up the problem.